src/util/http.ts

```diff
@@ -0,0 +1,25 @@
+import { parse } from 'url';
+import { IncomingMessage } from 'http';
+
+export const fetch = async (url: string): Promise<any> => {
+    const proto = parse(url).protocol.slice(0, -1);
+    const fetcher = await import(proto);
+    return await new Promise<string>(async (resolve, reject) => {
+        const request = fetcher.get(url, (response: IncomingMessage) => {
+            if (response.statusCode < 200 || response.statusCode > 299) {
+                reject(new Error(`Failed to load service at: ${url}, status code:${response.statusCode}`));
```

Comment:
This is being used for more than services, so make the error message more generic.